### PR TITLE
Jay/dev plus ut

### PR DESCRIPTION
**Pull request by mritunjaykumar**
- Transition of POC code to feature based code
- Removed dead and unused functionality code
- Added Unit Test Cases

### CODE REVIEW

File: src/test/java/com/rackspacecloud/metrics/influxdbscaler/InfluxDBHelperTests.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	private final static String SHOW_STATS = "q=SHOW STATS";
+
+	RestTemplate restTemplateMock;
+
+	@MockBean
+	StatefulSetProvider statefulSetProviderMock;
+
+	@Before
+	public void setUp() {
+		MockitoAnnotations.initMocks(this);
```
Comment (itzg) [inline src/test/java/com/rackspacecloud/metrics/influxdbscaler/InfluxDBHelperTests.java]: The `SpringRunner` and `@SpringBootTest` already trigger the mock annotation processing.

File: src/test/java/com/rackspacecloud/metrics/influxdbscaler/InfluxDBHelperTests.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+		restTemplateMock = mock(RestTemplate.class);
```
Comment (itzg) [inline src/test/java/com/rackspacecloud/metrics/influxdbscaler/InfluxDBHelperTests.java]: If you add `@Mock` to the field you'll get the mock created for you.

File: src/test/java/com/rackspacecloud/metrics/influxdbscaler/InfluxDBHelperTests.java (hunk 1 above)
Comment (mritunjaykumar) [reply src/test/java/com/rackspacecloud/metrics/influxdbscaler/InfluxDBHelperTests.java]: Done

File: src/test/java/com/rackspacecloud/metrics/influxdbscaler/InfluxDBHelperTests.java (hunk 2 above)
Comment (mritunjaykumar) [reply src/test/java/com/rackspacecloud/metrics/influxdbscaler/InfluxDBHelperTests.java]: Done

**Pull request merged**